libyear-gradle-plugin/src/main/kotlin/com/libyear/traversal/DependencyTraversal.kt

```diff
@@ -47,15 +43,50 @@ class DependencyTraversal private constructor(
     }
   }
 
+  @VisibleForTesting
+  fun shouldIncludeModule(
+    module: String,
+    depth: Int
+  ): Boolean {
+    if (maxTransitiveDepth != null && depth > maxTransitiveDepth) {
+      return false
+    }
+
+    // Inclusions supersede exclusions
+    val matchedInclusion = includedPatterns.firstOrNull { pattern -> pattern.second.matches(module) }
+    val matchedExclusion = excludedPatterns.firstOrNull { pattern -> pattern.second.matches(module) }
+
+    if(matchedInclusion != null) {
+      logger.info("Including $module because it matches ${matchedInclusion.first}")
+    } else if(matchedExclusion != null) {
+      logger.info("Excluding $module because it matches ${matchedExclusion.first}")
+      return false
+    }
+
+    return true
+  }
+
   companion object {
 
     fun visit(
       logger: Logger,
       root: ResolvedComponentResult,
-      logger: Logger,
       visitor: DependencyVisitor,
       maxTransitiveDepth: Int? = null,
-      excludedModules: Set<String> = emptySet()
-    ): Unit = DependencyTraversal(logger, visitor, maxTransitiveDepth, excludedModules).visit(root, depth = 0)
+      excludedModules: Set<String> = emptySet(),
+      includedModules: Set<String> = emptySet()
+    ): Unit = DependencyTraversal(
+      logger,
+      visitor,
+      maxTransitiveDepth,
+      excludedModules,
+      includedModules
+    ).visit(root, depth = 0)
+
+    @VisibleForTesting
+    fun String.wildcardToRegex(): Regex {
+      val globsToRegex = this.replace(".", "\\.").replace("*", ".*")
```

Comment:
Since this is regex under the hood I suppose a discerning Regex user could just cram their regex use case into the include/exclude list, so long as it doesn't actually use `.` 😆